### PR TITLE
[rush] Add --from-version-policy build flag  #1494

### PR DESCRIPTION
**Pull request by domarmstrong**
A a mirror of --to-version-policy flag the same way --to has --from. #1494

### CODE REVIEW

File: common/changes/@microsoft/rush/master_2019-09-06-22-37.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Add --from-version-policy flag for \"rush rebuild\" to allow building from a version policy with direct and indirect dependant projects",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/master_2019-09-06-22-37.json]:
```suggestion
      "comment": "Add a --from-version-policy option for bulk commands to allow running the command (like build) from packages with a version policy and their direct and indirect dependent projects",
```

File: apps/rush-lib/src/cli/scriptActions/BulkScriptAction.ts [hunk 2]
```diff
@@ -156,6 +157,12 @@ export class BulkScriptAction extends BaseScriptAction {
       argumentName: 'PROJECT1',
       description: 'Run command in the specified project and all of its dependencies'
     });
+    this._fromVersionPolicy =  this.defineStringListParameter({
+      parameterLongName: '--from-version-policy',
+      argumentName: 'VERSION_POLICY_NAME',
+      description: 'Run command in all projects with the specified version policy '
+        + 'and all projects that directly or indirectly depend on them'
```
Comment (iclanton) [inline apps/rush-lib/src/cli/scriptActions/BulkScriptAction.ts]:
```suggestion
        + 'and all projects that directly or indirectly depend on projects with the specified version policy'
```

File: apps/rush-lib/src/cli/scriptActions/BulkScriptAction.ts [hunk 3]
```diff
@@ -184,11 +191,13 @@ export class BulkScriptAction extends BaseScriptAction {
...
   }
 
-  private _mergeToProjects(): string[] {
-    const projects: string[] = [...this._toFlag.values];
-    if (this._toVersionPolicy.values && this._toVersionPolicy.values.length) {
+  private _mergeProjectsWithVersionPolicy(flags: CommandLineStringListParameter,
+    versionPolicies: CommandLineStringListParameter): string[] {
+
+    const projects: string[] = [...flags.values];
+    if (versionPolicies.values && versionPolicies.values.length) {
```
Comment (iclanton) [inline apps/rush-lib/src/cli/scriptActions/BulkScriptAction.ts]: Let's make this check more explicit.

```suggestion
    if (versionPolicies.values && versionPolicies.values.length > 0) {
```

Comment (domarmstrong) [reply apps/rush-lib/src/cli/scriptActions/BulkScriptAction.ts]: 👍

**Comment by domarmstrong** [comment]
Should I squash these, or do you keep the full history?

**Comment by iclanton** [comment]
Keeping the full history is fine.

**Pull request merged**